Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/FlightController/Compass/FCCompassViewController.swift

```diff
@@ -16,79 +16,58 @@ class FCCompassViewController: DJIBaseViewController, DJIFlightControllerDelegat
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view from its nib.
-        let fc: DJIFlightController? = self.fetchFlightController()
-        if fc != nil {
-            fc!.delegate = self
+        if let fc = self.fetchFlightController() {
+            fc.delegate = self
         }
     }
     
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
-    }
-    
     @IBAction func onCompassCalibrationButtonClicked(sender: UIButton) {
         let STOP_TAG: Int = 100
         let START_TAG: Int = 101
-        let fc: DJIFlightController? = self.fetchFlightController()
-        if fc != nil {
-            if sender.tag == STOP_TAG {
-                fc?.compass?.stopCalibrationWithCompletion({[weak self](error: NSError?) -> Void in
-                    if error != nil {
-                        self?.showAlertResult("Stop Calibration:\(error?.localizedDescription)")
-                    }
-                    else {
-                        sender.setTitle("Start Calibration", forState: .Normal)
-                        sender.tag = START_TAG
-                    }
-                })
-            }
-            else {
-                fc?.compass?.startCalibrationWithCompletion({[weak self](error: NSError?) -> Void in
-                    if error != nil {
-                        self?.showAlertResult("Start Calibration:\(error?.localizedDescription)")
-                    }
-                    else {
-                        sender.setTitle("Stop Calibration", forState: .Normal)
-                        sender.tag = STOP_TAG
-                    }
-                })
+        guard let fc = self.fetchFlightController() else {
+            self.showAlertResult("Component Not Exist.")
+            return
+        }
+        if sender.tag == STOP_TAG {
+            fc.compass?.stopCalibrationWithCompletion { [weak self] (error: NSError?) -> Void in
+                if let error = error {
```

Comment:
This kind of constant capturing is very confusion because of the same names and different scopes. 
Consider using different names here for clarity.